frappe/model/base_document.py

```diff
@@ -969,7 +969,7 @@ def get_value(self, fieldname):
 		return self.cast(val, df)
 
 	def cast(self, value, df):
-		return cast_fieldtype(df.fieldtype, value)
+		return cast(df.fieldtype, value)
```

Comment:
Other than this change... LGTM.

The previous behaviour of cast_fieldtype (`None => current_time`) might have been used while writing business logic sadly. So this becomes a breaking change and could introduce bugs 😅  

On the flip side, expecting `None` to evaluate to current time instead of just using `now()`, `nowdate()` etc is just as bad. But ... 🤷 

https://github.com/frappe/frappe/blob/1ac27efa7a252701e4c7c61c8aacc90fe401cb99/frappe/utils/data.py#L49-L53